evalml/data_checks/target_leakage_data_check.py

```diff
@@ -63,14 +65,28 @@ def validate(self, X, y):
         y = _convert_to_woodwork_structure(y)
         X = _convert_woodwork_types_wrapper(X.to_dataframe())
         y = _convert_woodwork_types_wrapper(y.to_series())
+        if not self.pearson:
+            highly_corr_cols = []
+            target = 'target'
+            while target in X.columns:
```

Comment:
Do we need to do this? I like that we're extracting out the column from the DF and comparing it to the target but if mutual information is greater than the threshold for that [target, col], we should know what the col is without labeling the target this way, right?

Reply:
@angela97lin I added this so that I could add the target column to the X dataframe without any column name conflicts. This way, I can iterate through the woodwork DataTable columns and easily pick out the column of interest and the target column. Was there a better way to do this?

Reply:
@bchen1116 I don't think you need to append the target column to the X dataframe in order to iterate if we choose to build a dataframe at each iteration with the column of interest and the target column. Does that make sense?

Reply:
Left a comment, but this was just how I avoided naming conflicts.

Reply:
@angela97lin changed the implementation to avoid this. Thanks for the input, I think it's much cleaner now!